e2e/helpers/environment/environment-factory.ts

```diff
@@ -0,0 +1,60 @@
+import Docker from 'dockerode';
+import baseDebug from '@tryghost/debug';
+import {DEV_ENVIRONMENT} from './constants';
+import {DevEnvironmentManager} from './dev-environment-manager';
+import {EnvironmentManager} from './environment-manager';
+
+const debug = baseDebug('e2e:EnvironmentFactory');
+
+// Cached manager instance (one per worker process)
+let cachedManager: EnvironmentManager | DevEnvironmentManager | null = null;
+
+/**
+ * Check if the dev environment (yarn dev:forward) is running.
+ * Detects by checking for the ghost_dev network and running MySQL container.
+ */
+export async function isDevEnvironmentAvailable(): Promise<boolean> {
```

Comment:
Do you find this way sufficient to check whether dev environment is running locally?